server/routes/nl_interface_next.py

```diff
@@ -291,6 +289,11 @@ def page():
   if (os.environ.get('FLASK_ENV') == 'production' or
       not current_app.config['NL_MODEL']):
     flask.abort(404)
+  # For the NL module, launch classifier training. This needs to happen here
```

Comment:
@shifucun This is pretty inelegant. Is there any other place I can call this function train_classifiers() AFTER the current_app is already created and running? The API calls in the train_classifiers() (to get embeddings vectors from the NL embeddings server) need the current_app context.

Reply:
you can do this in main.py after create_app(). But I wonder why this does not belong to the nl server?

Reply:
I did try to use it in main after create_app() but it still told me that when using the services.datacommons functions for the API calls that the app context was not there. Basically, these lines fail in the functions in services.datacommons:

url = f'{cfg.NL_ROOT}/api/embedding?q={query}'
return get(url).get('embeddings_vector', [])

I suspect that it is after app.run() is called that these API calls can then be made. So that's why I have put them in the nl_interface.py routes which is not ideal. Thoughts?

Re: why this does not belong in the nl server, the reason is that these are for query classifications which are currently not planned to be in the nl_server. We can eventually do that too but I think right now we were planning to keep them in the main website. We can discuss with the team if we want to use the NL server for these as well? Eventually, I agree. But for now, probably not?

Reply:
We can submit as is and I can take a look afterwards.

Reply:
ok thanks!